### PR TITLE
Factor mono_get_exception_missing_field and mono_get_exception_missin…

### PR DESCRIPTION
**Pull request by jaykrell**
…g_method into mono_get_exception_missing_member ahead of coop conversion.

### CODE REVIEW

File: mono/metadata/exception.c [hunk 1]
```diff
@@ -595,13 +595,14 @@ mono_get_exception_not_supported (const char *msg)
...
+ * mono_get_exception_missing_member:
+ * \param member_type field or method
  * \param class_name the class where the lookup was performed.
  * \param member_name the name of the missing method.
  * \returns a new instance of the \c System.MissingMethodException
  */
 MonoException *
-mono_get_exception_missing_method (const char *class_name, const char *member_name)
+mono_get_exception_missing_member (const char *member_type, const char *class_name, const char *member_name)
 {
```
Comment (vargaz) [inline mono/metadata/exception.c]: This is actually not the member type but the exception type.

File: mono/metadata/exception.c [hunk 2]
```diff
@@ -595,13 +595,14 @@ mono_get_exception_not_supported (const char *msg)
 }
 
 /**
- * mono_get_exception_missing_method:
+ * mono_get_exception_missing_member:
+ * \param member_type field or method
  * \param class_name the class where the lookup was performed.
  * \param member_name the name of the missing method.
  * \returns a new instance of the \c System.MissingMethodException
  */
```
Comment (vargaz) [inline mono/metadata/exception.c]: This comment is now obsolete.

**Comment by jaykrell** [comment]
Well, perhaps with variance as to ignoring or asserting on errors.

**Comment by jaykrell** [comment]
Oh I mixed up const char* and MonoString, making this PR better.

**Comment by jaykrell** [comment]
Flaky tests.

**Comment by jaykrell** [comment]
@monojenkins buid failed

**Pull request merged**